Packages/Tracking/Physical Hands/Runtime/Scripts/ContactHand.cs

```diff
@@ -0,0 +1,420 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2023.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.PhysicalHands
+{
+    public abstract class ContactHand : MonoBehaviour
+    {
+        internal const int FINGERS = 5, FINGER_BONES = 3;
+
+        internal ContactBone[] bones;
+        internal ContactBone palmBone;
+
+        internal Chirality _handedness = Chirality.Left;
+        public Chirality Handedness => _handedness;
+
+        private Hand modifiedHand = new Hand();
+        internal Hand dataHand = new Hand();
+        [SerializeField]
+        internal bool tracked = false, resetting = false, ghosted = false;
+        internal bool isHandPhysical = true;
+        /// <summary>
+        /// Is the hand tracked from a visual sense?
+        /// </summary>
+        public bool Tracked => tracked;
+        /// <summary>
+        /// Is the hand able to interact with the physic simulation, or is it in a reset state?
+        /// </summary>
+        public bool Ghosted => ghosted;
+
+        /// <summary>
+        /// Is the hand going to apply physics forces or should it pass through objects?
+        /// </summary>
+        public bool IsHandPhysical => isHandPhysical;
```

Comment:
adjust summary to explain what IsHandPhysical actually does

consider renaming IsHandPhysical